src/main/java/com/mraof/minestuck/world/lands/terrain/EndLandType.java

```diff
@@ -85,21 +81,28 @@ public void setProperties(LandProperties properties)
 	@Override
 	public void setBiomeSettings(LandWrapperBiome biome, StructureBlockRegistry blocks)
 	{
-		if(biome.type == BiomeType.NORMAL)
+		if(biome.type != BiomeType.OCEAN)
 		{
 			biome.addFeature(GenerationStage.Decoration.VEGETAL_DECORATION, MSFeatures.END_TREE.withConfiguration(EndTree.getDefaultConfig().build()).withPlacement(Placement.COUNT_EXTRA_HEIGHTMAP.configure(new AtSurfaceWithExtraConfig(2, 0.1F, 1))));
+		}
+		
+		if(biome.type == BiomeType.NORMAL)
+		{
+			biome.addFeature(GenerationStage.Decoration.TOP_LAYER_MODIFICATION, MSFeatures.SURFACE_DISK.withConfiguration(new SphereReplaceConfig(MSBlocks.END_GRASS_BLOCK.getDefaultState(), 5, 1, Lists.newArrayList(blocks.getBlockState("surface"), MSBlocks.END_GRASS_BLOCK.getDefaultState()))).withPlacement(Placement.COUNT_TOP_SOLID.configure(new FrequencyConfig(1))));
```

Comment:
Because it's placing a grass block, it might be more appropriate to make a feature that strictly places blocks if they have an air block/non-solid above them (or as an alternative, gets its positions for placing blocks as the topmost block in an xz-area), since the surface block is not just used for the surface. Though I suppose it will only matter in some unusual cases as long as the upper blocks differ from the surface blocks. So if you choose to leave it be, make sure to leave a TODO-comment about it.

Reply:
I have not noticed any major issues with this so far, but it is definitely worth making the change

Reply:
I cannot see a way to check for both the given block and if the block above it is air within the SphereReplaceConfig, I may leave this as a TODO

Reply:
With the surface block now being the same as the upper block, this has become more important. Also, it wouldn't happen in SphereReplaceConfig but rather in a feature class. For example a class similar to SurfaceDiskFeature, or in SurfaceDiskFeature but with a boolean parameter like PillarFeature.

Reply:
Changes made! Appears to work well